### ISSUE TITLE
False negative with "promise/catch-or-return"

### ISSUE THREAD

**Issue by torarnek** [xjamundx/eslint-plugin-promise#22]
I get a false negative with the following code, complaining with "Expected catch() or return  promise/catch-or-return":

```javascript
postJSON("/smajobber/api/reportJob.json", { jobId: this.props.jobId})
                .then(() => this.setState({f: true}))
                .catch(() => this.setState({f: true}));
```
Where PostJSON return a Promise.

**Comment by xjamundx** [comment]
weird that shouldn't error. let me see if i can figure it out.

a PR with a failing test might be helpful too if you have time.

**Comment by joeybaker** [comment]
Here's a test case you can use:

This code shows an error for me:
```js
return myPromise.then((result) => {
  if (result.truthy) return Promise.resolve()
  else return Promise.resolve()
})
```

**Comment by xjamundx** [comment]
Thanks @joeybaker will try to fix this!

**Comment by ahuff44** [comment]
That looks like #18 to me @joeybaker; it's a bug with promise/always-return, not promise/catch-or-return

**Comment by joeybaker** [comment]
You're probably right @ahuff44!

**Comment by xjamundx** [comment]
Going to take a look at this today. May have been fixed in the latest `catch-or-return` update, but will verify and respond.

**Comment by xjamundx** [comment]
yeah I can't reproduce this error in the latest version of eslint-plugin-promise (3.0.0), feel free to reopen if you're still seeing a problem!
